### PR TITLE
Revise the APIs

### LINKED ISSUE #3: ReConsider the APIs
First of all, thank you for creating such a nice library.

I reconsider the pixela_ex apis and found some changes make it better.
However it includes breaking changes, so I propose you changes before implement.

If you feel OK, I will implement this.
Any thoughts?

To make api convenient, the changes follows following principles:
- When it was finished without errors, it returns `{:ok, ____}`. Otherwise it returns `{:error, ___}`
- required parameters bring as arguments
- optional parameters bring as an argument of [Keyword](https://hexdocs.pm/elixir/Keyword.html)

| propose to change | current | new | note |
| - | - | - | - |
| o | `create_user(%{username: username, token: token, agree_terms_of_service: agree_terms_of_service, not_minor: not_minor})` | `create_user(username, token_will_be_used, agree_terms_of_service, not_minor)` | https://pixe.la/#api-detail-post-users |
| o | `update_user(username, token, %{new_token: new_token})` | `update_user(username, token, new_token)` | https://pixe.la/#api-detail-put-user |
| | `delete_user(username, token)` | `delete_user(username, token)` | https://pixe.la/#api-detail-delete-user |
| o | `create_graph(username, token, %{graph_id: graph_id, name: name, unit: unit, type: quantity_type, color: color})` | `create_graph(username, token, id, name, unit, type, color)` | https://pixe.la/#api-detail-post-graphs |
| | `get_graphs(username, token)` | `get_graphs(username, token)` | https://pixe.la/#api-detail-get-graphs |
| o | `graph_url(username, graph_id, %{date: date, mode: mode})` | `get_graph(username, graph_id, date: date, mode: mode)` | https://pixe.la/#api-detail-get-graph |
| o | `update_graph(username, token, graph_id, %{name: name, unit: unit, color: color})` | `update_graph(username, token, graph_id, name: name, unit: unit, color: color, purge_cache_urls: purge_cache_urls)` | https://pixe.la/#api-detail-put-graph |
| | `delete_graph(username, token, graph_id)` | `delete_graph(username, token, graph_id)` | https://pixe.la/#api-detail-delete-graph |
| o | `create_pixel(username, token, graph_id, %{date: date, quantity: quantity})` | `create_pixel(username, token, graph_id, date, quantity)` | https://pixe.la/#api-detail-post-pixels |
| | `get_pixel(username, token, graph_id, date)` | `get_pixel(username, token, graph_id, date)` | https://pixe.la/#api-detail-get-pixel |
| o | `update_pixel(username, token, graph_id, date, %{quantity: quantity})` | `update_pixel(username, token, graph_id, date, quantity)` | https://pixe.la/#api-detail-put-pixel |
| | `increment_pixel(username, token, graph_id)` | `increment_pixel(username, token, graph_id)` | https://pixe.la/#api-detail-pixel-increment |
| | `decrement_pixel(username, token, graph_id)` | `decrement_pixel(username, token, graph_id)` | https://pixe.la/#api-detail-pixel-decrement |
| | `delete_pixel(username, token, graph_id, date)` | `delete_pixel(username, token, graph_id, date)` | https://pixe.la/#api-detail-delete-pixel | 
| o | `create_webhook(username, token, %{graph_id: graph_id, type: countup_type})` | `create_webhook(username, token, graph_id, type)` | https://pixe.la/#api-detail-post-webhooks |
| | `get_webhooks(username, token)` | `get_webhooks(username, token)` | https://pixe.la/#api-detail-get-webhooks |
| | `invoke_webhook(username, webhook_hash)` | `invoke_webhook(username, webhook_hash)` | https://pixe.la/#api-detail-post-webhook |
| | `delete_webhook(username, token, webhook_hash)` | `delete_webhook(username, token, webhook_hash)` | https://pixe.la/#api-detail-delete-webhook |

### PR DESCRIPTION
**Pull request by niku**
fixes #3

### CODE REVIEW

File: lib/pixela_ex/client/graph_functions.ex [hunk 1]
```diff
@@ -44,19 +44,19 @@ defmodule PixelaEx.Client.GraphFunctions do
...
 
-      iex> PixelaEx.Client.GraphFunctions.graph_url("a-know", "test-graph", %{date: "20180331"})
-      "https://pixe.la/v1/users/a-know/graphs/test-graph?date=20180331"
+      iex> PixelaEx.Client.GraphFunctions.get_graph("a-know", "test-graph", date: "20180331", mode: "short")
+      {:get, ["users/a-know/graphs/test-graph?date=20180331&mode=short", []]}
 
   """
-  @spec graph_url(PixelaEx.username, PixelaEx.graph_id, %{optional(:date) => PixelaEx.date, optional(:mode) => PixelaEx.mode}) :: String.t
-  def graph_url(username, graph_id, param \\ %{}) when is_map(param) do
+  @spec get_graph(PixelaEx.username, PixelaEx.graph_id, keyword(String.t)) :: String.t
```
Comment (otoyo) [inline lib/pixela_ex/client/graph_functions.ex]: Please use `[key: value_type, ...]` for available keys.
```
[date: PixelaEx.date, mode: PixelaEx.mode]
```

File: lib/pixela_ex/client/graph_functions.ex [hunk 2]
```diff
@@ -68,25 +68,26 @@ defmodule PixelaEx.Client.GraphFunctions do
...
 
   @doc """
   Update predefined pixelation graph definitions.
 
   ## Examples
 
-      iex> PixelaEx.Client.GraphFunctions.update_graph("a-know", "thisissecret", "test-graph", %{name: "graph-name", unit: "commit", color: "shibafu"})
-      {:put, ["users/a-know/graphs/test-graph", [body: %{name: "graph-name", unit: "commit", color: "shibafu"}, headers: ["X-USER-TOKEN": "thisissecret"]]]}
+      iex> PixelaEx.Client.GraphFunctions.update_graph("a-know", "thisissecret", "test-graph", name: "graph-name", unit: "commit", color: "shibafu", purge_cache_urls: ["https://camo.githubusercontent.com/xxx/xxxx"])
+      {:put, ["users/a-know/graphs/test-graph", [body: %{name: "graph-name", unit: "commit", color: "shibafu", purge_cache_urls: ["https://camo.githubusercontent.com/xxx/xxxx"]}, headers: ["X-USER-TOKEN": "thisissecret"]]]}
```
Comment (otoyo) [inline lib/pixela_ex/client/graph_functions.ex:80]: Thank you for adding `purge_cache_urls`!

File: lib/pixela_ex/client/graph_functions.ex [hunk 3]
```diff
@@ -68,25 +68,26 @@ defmodule PixelaEx.Client.GraphFunctions do
...
 
   """
-  @spec update_graph(PixelaEx.username, PixelaEx.token, PixelaEx.graph_id, %{required(:name) => PixelaEx.name, required(:unit) => PixelaEx.unit, required(:color) => PixelaEx.color}) :: PixelaEx.http_result
-  def update_graph(username, token, graph_id, %{name: name, unit: unit, color: color}) do
-    body = %{
-      name:   name,
-      unit:   unit,
-      color:  color
-    }
+  @spec update_graph(PixelaEx.username, PixelaEx.token, PixelaEx.graph_id, keyword) :: PixelaEx.http_result
```
Comment (otoyo) [inline lib/pixela_ex/client/graph_functions.ex]: `[purge_cache_urls: [String.t]]` preferred instead of `keyword`

**Review by otoyo — COMMENTED** [review]
Thank you for sending this PR.
I commented some.

**Comment by niku** [comment]
@otoyo thank you for your reviews ❤️  I updated the codes.

**Review by otoyo — APPROVED** [review]
Thank you for update!

**Comment by otoyo** [comment]
Now I released v1.0.0
https://github.com/otoyo/pixela_ex/releases/tag/1.0.0

Thank you for your contributions!

**Pull request merged**